src/lib/comms/include/sol-http.h

```diff
@@ -236,9 +250,108 @@ sol_http_param_init(struct sol_http_param *params)
 
 bool sol_http_param_add(struct sol_http_param *params,
     struct sol_http_param_value value) SOL_ATTR_WARN_UNUSED_RESULT;
+bool sol_http_param_add_copy(struct sol_http_param *params, struct sol_http_param_value value);
 void sol_http_param_free(struct sol_http_param *params);
 
-int sol_http_escape_string(char **escaped, const char *value);
+/**
+ * Encodes an URL string.
+ *
+ * If the string was not encoded this function will simple copy
+ * the slices content to the buffer and set @c SOL_BUFFER_FLAGS_MEMORY_NOT_OWNE.D
+ *
+ * @param buf The buffer that will hold the encoded string.
```

Comment:
Say this buffer is being initialized in this function. If buffer contained some data, it will be lost as no `sol_buffer_fini()` is lost.

Make it clear the resulting buffer, if no encode was needed, will point to value.data and the flag will be set to SOL_BUFFER_FLAGS_MEMORY_NOT_OWNED, as you mentioned in the description of the function. Say that buffer should be finalized with `sol_buffer_fini()` and that if a copy is needed, users should call `sol_buffer_steal_or_copy()`.